File: Python/bubblesort.py
def bubblesort(arr):
    n=len(arr)
    #print(n)
    for i in range (n):
        swapped=False
        for j in range(1,n-i):
            #print(j)
            if (arr[j-1]>arr[j]):
                swapped=True
                swap(arr,j,j-1)
        #print()
        if (not swapped):
            break

def swap(arr,a,b):
    arr[a],arr[b]=arr[b],arr[a]

def main():
    arr=[5,1,2,3,4,0]
    bubblesort(arr)
    print(arr)

if __name__=="__main__":
    main()

File: C/bubblesort.c
#include <stdio.h>

void swap(int arr[], int a, int b)
{
    int temp = arr[a];
    arr[a] = arr[b];
    arr[b] = temp;
}

void bubblesort(int arr[], int size)
{
    // int count = sizeof(arr) / sizeof(arr[0]);
    // printf("%d%d%d", count, sizeof(arr), sizeof(arr[0]));
    for (int i = 0; i < size; i++)
    {
        int flag = 0;
        for (int j = 1; j < size - i; j++)
        {
            // printf("%d", j);
            if (arr[j - 1] > arr[j])
            {
                swap(arr, j, j - 1);
                flag = 1;
            }
        }
        if (flag == 0)
        {
            break;
        }
        // printf("----\n");
    }
}

int main()
{
    printf("Hello");
    int arr[7] = {3, 4, 5, 6, 0, 9, 8};
    int count = sizeof(arr) / sizeof(arr[0]);
    bubblesort(arr, count);
    // printf("Im main: %d%d%d end", count, sizeof(arr), sizeof(arr[0]));
    for (int i = 0; i < count; i++)
    {
        printf("%d", arr[i]);
    }
    return 0;
}